aea/protocols/generator/base.py

```diff
@@ -1772,8 +1772,13 @@ def _protocol_buffer_schema_str(self) -> str:
 
         # heading
         proto_buff_schema_str = self.indent + 'syntax = "proto3";\n\n'
-        proto_buff_schema_str += self.indent + "package aea.{}.{};\n\n".format(
-            self.protocol_specification.author, self.protocol_specification.name
+        proto_buff_schema_str += self.indent + "package aea.{}.{}.{};\n\n".format(
+            self.protocol_specification.protocol_specification_id.author,
+            self.protocol_specification.protocol_specification_id.name,
+            "v"
+            + self.protocol_specification.protocol_specification_id.version.replace(
+                ".", "_"
```

Comment:
this should be defined in a helper method somewhere for reuse. Could extend public id and introduce `to_undotted` or something like this

Reply:
(also `from_undotted` so we have these for reusability)